test/semantics/symbol06.f90

```diff
@@ -97,7 +97,7 @@ subroutine s1
  i = x%t1
  !REF: /s1/i
  !REF: /s1/x
- !DEF: /s1/t3/t2 ObjectEntity TYPE(t2)
+ !DEF: /s1/t3/t2 ObjectEntity TYPE(t1)
```

Comment:
I don't understand this change. t3 has an implicit member named t2 whose type is type(t2).

Reply:
t2 is the local alias for the use-associated type name t1, and the code that adds the parent component to t3 ends up working with the fully resolved type rather than the local name.